library/core/src/num/dec2flt/common.rs

```diff
@@ -0,0 +1,220 @@
+//! Common utilities, for internal use only.
+
+use crate::ptr;
+
+/// Helper methods to process immutable bytes.
+pub trait ByteSlice: AsRef<[u8]> {
```

Comment:
nit: you could remove quite a lot of these functions by having the trait bound be `Deref<[u8]>` instead.

Reply:
The issue is it needs to be implemented for `[u8]`, which correct me if I'm wrong, there's no good way to implement `Deref<Target=[u8]>` for `[u8]` without infinite recursion?

Reply:
&T implements Deref<Target = T> for all T.

Reply:
Same issue with implementing this as [here](https://github.com/rust-lang/rust/pull/86761#discussion_r663408617). Let me know if you have any suggestions or this has been insufficiently resolved.

Reply:
Fair enough - I took a crack at it and it was indeed rather tricky.